rust/lance/src/dataset/write/merge_insert.rs

```diff
@@ -3138,4 +3241,405 @@ mod tests {
             StreamingTableExec: partition_sizes=1, projection=[value, key]"
         ).await.unwrap();
     }
+
+    #[tokio::test]
+    async fn test_explain_plan() {
+        // Set up test data
+        let schema = Arc::new(Schema::new(vec![
+            Field::new("id", DataType::Int32, false),
+            Field::new("name", DataType::Utf8, true),
+        ]));
+
+        let batch = RecordBatch::try_new(
+            schema.clone(),
+            vec![
+                Arc::new(Int32Array::from(vec![1, 2, 3])),
+                Arc::new(StringArray::from(vec!["a", "b", "c"])),
+            ],
+        )
+        .unwrap();
+
+        let dataset = Dataset::write(
+            RecordBatchIterator::new(vec![Ok(batch.clone())], schema.clone()),
+            "memory://test_explain_plan",
+            None,
+        )
+        .await
+        .unwrap();
```

Comment:
For brevity, use `lance_datagen` to create the test dataset. Here's an example from another test:

https://github.com/lancedb/lance/blob/5d72cbf6b6f147ff09169fb86c0cb8b24933546f/rust/lance/src/dataset/optimize.rs#L2809-L2817